code/game/machinery/autodoc_excelsior.dm

```diff
@@ -0,0 +1,249 @@
+	update_icon()
+	
+
+/obj/machinery/excelsior_autodoc/Destroy()
+	if(occupant)
+		qdel(occupant)
+	return ..()
+
+/obj/machinery/excelsior_autodoc/relaymove(mob/user)
+	if (user.stat)
+		return
+	src.go_out()
+	return
+
+/obj/machinery/excelsior_autodoc/attackby(obj/item/I, mob/living/user)
+	if(default_deconstruction(I, user))
+		return
+	if(default_part_replacement(I, user))
+		return
+	..()
+
+/obj/machinery/excelsior_autodoc/verb/eject()
+	set src in view(1)
+	set category = "Object"
+	set name = "Eject Occupant"
+
+	if (usr.incapacitated())
+		return
+	src.go_out()
+	add_fingerprint(usr)
+	return
+
+/obj/machinery/excelsior_autodoc/verb/move_inside()
+	set src in view(1)
+	set category = "Object"
+	set name = "Enter Autodoc"
+
+	if(usr.stat)
+		return
+	if(src.occupant)
+		to_chat(usr, SPAN_WARNING("The autodoc is already occupied!"))
+		return
+	if(usr.abiotic())
+		to_chat(usr, SPAN_WARNING("The subject cannot have abiotic items on."))
+		return
+	set_occupant(usr)
+	src.add_fingerprint(usr)
+	return
+
+/obj/machinery/excelsior_autodoc/proc/go_out()
+	if (!occupant || cover_locked)
+		return
+	if(autodoc_processor.active)
+		to_chat(usr, SPAN_WARNING("Comrade, autodoc is locked down! Wait until all operations is done."))
+		return
+	if(cover_closed)
+		open_cover()
+
+	for(var/obj/O in src)
+		O.forceMove(loc)
+	occupant.forceMove(loc)
+	occupant.reset_view()
+	occupant.unset_machine()
+	occupant = null
+	autodoc_processor.set_patient(null)
+	update_use_power(1)
+	update_icon()
+
+/obj/machinery/excelsior_autodoc/proc/set_occupant(var/mob/living/L)
+	L.forceMove(src)
+	src.occupant = L
+	src.add_fingerprint(usr)
+	autodoc_processor.set_patient(L)
+	update_use_power(2)
+	L.set_machine(src)
+	cover_state = image(icon, "opened")
+	cover_state.layer = 4.5
+
+	if(!is_excelsior(L))
+		cover_locked = TRUE
+		close_cover()
+		sleep(20)
+		to_chat(L, SPAN_DANGER("Autodoc implanting you with something!"))
+		sleep(40)
+		var/obj/item/implant/excelsior/implant = new(L)
+		implant.install(L, BP_HEAD)
+		var/datum/faction/F = get_faction_by_id(FACTION_EXCELSIOR)
+		var/datum/objective/timed/excelsior/E = (locate(/datum/objective/timed/excelsior) in F.objectives)
+		if(E)
+			if(!E.active)
+				E.start_excel_timer()
+			else
+				E.on_convert()
+		cover_locked = FALSE
+	else
+		update_icon()
+
+/obj/machinery/excelsior_autodoc/attack_hand(mob/user)
+	if(occupant)
+		if(!cover_closed)
+			close_cover()
+
+		ui_interact(user)
+
+/obj/machinery/excelsior_autodoc/affect_grab(var/mob/user, var/mob/target)
+	if (src.occupant)
+		to_chat(user, SPAN_NOTICE("The autodoc is already occupied!"))
+		return
+	if(target.buckled)
+		to_chat(user, SPAN_NOTICE("Unbuckle the subject before attempting to move them."))
+		return
+	if(target.abiotic())
+		to_chat(user, SPAN_NOTICE("Subject cannot have abiotic items on."))
+		return
+	set_occupant(target)
+	src.add_fingerprint(user)
+	return TRUE
+
+/obj/machinery/excelsior_autodoc/MouseDrop_T(var/mob/target, var/mob/user)
```

Comment:
no need to do var/mob/target, mob/target does the same thing

Reply:
What's the difference then?